src/providers/realtime-provider.js

```diff
@@ -113,15 +167,29 @@ export default class RealtimeEventsProvider {
         this.socket.onmessage = (event) => {
             let data = JSON.parse(event.data);
 
-            if (data.type === MESSAGES.DATA_TYPE_REPLY) {
+            if (data.type === DATA_TYPES.DATA_TYPE_REPLY) {
                 let replyToId = data.data.replyTo;
                 let subscriptionDetails = this.getSubscriptionDetailsById(replyToId);
                 subscriptionDetails.call = data.call;
-            } else if (data.type === MESSAGES.DATA_TYPE_EVENTS) {
+            } else if (this.isSupportedDataType(data.type)) {
                 let call = data.call;
                 let subscriptionDetails = this.getSubscriptionDetailsByCall(call);
 
-                subscriptionDetails.callback(data.data);
+                if (data.type === DATA_TYPES.DATA_TYPE_EVENTS) {
```

Comment:
Do you think there's any value in creating sub-classes for each type - one for handling event data, another for telemetry data, maybe a third for imagery. Might improve readability and separate concerns.

Reply:
I would say yes if there were more differences between the types. The only differences are between events and everything else, from what I can see. In the realtime-provider is just a check for type events and a straight hit to the callback. I think most of the other differentiation is handled in messages.